src/e2e/app/common/src/e2e_harness.rs

```diff
@@ -0,0 +1,186 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::future::Future;
+
+use kamu_cli_e2e_common::KamuApiServerClient;
+use kamu_node_puppet::extensions::KamuNodePuppetExt;
+use kamu_node_puppet::{KamuNodePuppet, NewWorkspaceOptions};
+use regex::Regex;
+use sqlx::{PgPool, SqlitePool};
+
+use crate::{api_flight_sql_e2e_test, api_server_e2e_test, KamuFlightSQLClient};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Default)]
+pub struct KamuNodeApiServerHarnessOptions {
+    env_vars: Vec<(String, String)>,
+    kamu_config: Option<String>,
```

Comment:
here and above: `kamu_api_server_config`

Reply:
`kamu_api_server` is more like the name of the binary file we will be testing

(to avoid associations with `kamu` (CLI))